src/main/java/frc/robot/IMU/BNO055.java

```diff
@@ -0,0 +1,796 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.IMU;
+
+import java.util.TimerTask;
+import java.util.concurrent.atomic.AtomicReference;
+
+import edu.wpi.first.wpilibj.I2C;
+import edu.wpi.first.wpilibj.Timer;
+
+public class BNO055 {
+  // Tread variables
+  private java.util.Timer executor;
+  private static final long THREAD_PERIOD = 20; // ms - max poll rate on sensor.
+
+  public static final byte BNO055_ADDRESS_A = 0x28;
+  public static final byte BNO055_ADDRESS_B = 0x29;
+  public static final int BNO055_ID = 0xA0;
+
+  private static BNO055 instance;
+
+  private static I2C imu;
```

Comment:
`requestedMode` and `requestedVectorType` should be instance variables, not statics